File: solutions/vbnet/high-scores/1/HighScores.vb
Imports System
Imports System.Linq
Imports System.Collections.Generic

Public Class HighScores
    Private ReadOnly _list As List(Of Integer)

    Public Sub New(ByVal list As List(Of Integer))
        _list = list
    End Sub

    Public Function Scores() As List(Of Integer)
        Return _list
    End Function

    Public Function Latest() As Integer
        Return _list.Last()
    End Function

    Public Function PersonalBest() As Integer
        Return _list.Max()
    End Function

    Public Function PersonalTopThree() As List(Of Integer)
        Return _list.OrderByDescending(Function(score) score).Take(3).ToList()
    End Function

End Class

File: solutions/euphoria/high-scores/1/high-scores.ex
include std/math.e
include std/sort.e

public function latest(sequence scores)
  return scores[$]
end function

public function personal_best(sequence scores)
  return max(scores)
end function

public function personal_top_three(sequence scores)
  scores = sort(scores, stdsort:DESCENDING)
  if length(scores) > 3 then
    scores = scores[1..3]
  end if
  return scores
end function